sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3511,8 +3511,9 @@ object SQLConf {
     buildConf("spark.sql.inferTimestampNTZInDataSources.enabled")
       .doc("When true, the TimestampNTZ type is the prior choice of the schema inference " +
         "over built-in data sources. Otherwise, the inference result will be TimestampLTZ for " +
-        "backward compatibility. As a result, for JSON/CSV files written with TimestampNTZ " +
-        "columns, the inference results will still be of TimestampLTZ types.")
+        "backward compatibility. As a result, for JSON/CSV files and partition directories  " +
+        "written with TimestampNTZ columns, the inference results will still be of TimestampLTZ " +
+        "types.")
       .version("3.4.0")
       .booleanConf
       .createWithDefault(false)
```

Comment:
does it mean users can't do NTZ roundtrip (write and read) in 3.4 by default?

Reply:
Yup. Take partition directory naming as an example, the outputs from Timestamp NTZ and LTZ are exactly the same.